server/models/snippet.js

```diff
@@ -1,30 +1,62 @@
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
 
 const snippetSchema = new mongoose.Schema({
-	code: {
-		type: String,
-		required: true,
-		
-	},
-	author: {
-		type: mongoose.Schema.Types.ObjectId,
-		required: true,
-		ref: "User",
-	},
-	reviewer: {
-		type: mongoose.Schema.Types.ObjectId,
-		ref: "User",
-	},
-	title: {
-		type: String,
-		required: true,
-		default: "Unknown",
-	},
-	language: {
-		type: String,
-		required: true,
-	},
+  code: {
+    type: String,
+    required: true,
+  },
+  type: {
+    type: String,
+    required: true,
+    enum: ['request', 'receive'],
+  },
+  status: {
+    type: String,
+    enum: ['pending', 'accepted', 'completed'],
+    default: 'pending',
+  },
+  author: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: 'User',
+  },
+  reviewer: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  title: {
+    type: String,
+    required: true,
+    default: 'Unknown',
+  },
+  language: {
+    type: String,
+    required: true,
+  },
+  comments: {
+    type: String,
+    required: true,
+  },
+  rating: {
+    type: Number,
+    min: 0,
+    max: 5,
+  },
+  date_requested: {
+    type: Date,
+    required: true,
+    default: Date.now,
+  },
+  date_submitted: {
+    type: Date,
+    required: true,
+    default: Date.now,
+  },
+  date_accepted: {
+    type: Date,
+    required: true,
+  },
```

Comment:
- Should `comments` and `date_accepted` be `required`?

- What is the purpose of `date_submitted` field? Does it mean the date a review is submitted?

Reply:
I just follow the ticket to build the model. I suppose date_submitted is the date the reviewer submits the review. Do you think of any word more meaningful?

Reply:
I changed the comments and date_accepted to not required as well for our convenience. I thought comments include the feedback of reviewer, so it's compulsory. But now I deleted them first

Reply:
You're right I just checked the ticket. But I think this name is a bit vague. Maybe date_review_submitted?

Reply:
I think we keep it like this for now. I don't want the field name contains 3 words. Now I don't think of the better name for it. We can imply that 'submitted' means completion of the review